open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/utils/ConsentRetrievalUtil.java

```diff
@@ -256,4 +257,427 @@ public static JSONArray appendDummyAccountID() {
+                    dateTimeControlParameterArray);
+            consentDataJSON.add(jsonElementValidFromDateTime);
+        }
+
+        String maxAmount = controlParameters.getAsString(ConsentExtensionConstants.MAXIMUM_INDIVIDUAL_AMOUNT);
+        if (maxAmount != null) {
+            // Constructing jsonElementControlParameter
+            JSONObject jsonElementControlParameter = new JSONObject();
+            jsonElementControlParameter.appendField(ConsentExtensionConstants.TITLE,
+                    ConsentExtensionConstants.CONTROL_PARAMETER_MAX_INDIVIDUAL_AMOUNT_TITLE);
+            JSONArray controlParameterArray = new JSONArray();
+
+
+            JSONObject maximumIndividualAmount = ((JSONObject) controlParameters.
+                    get(ConsentExtensionConstants.MAXIMUM_INDIVIDUAL_AMOUNT));
+
+            String formattedAmount = String.format("%s %s",
+                    (maximumIndividualAmount.get(ConsentExtensionConstants.CURRENCY)),
+                    (maximumIndividualAmount.get(ConsentExtensionConstants.AMOUNT)));
+            controlParameterArray.add(formattedAmount);
+            jsonElementControlParameter.appendField(ConsentExtensionConstants.DATA_SIMPLE, controlParameterArray);
+
+            consentDataJSON.add(jsonElementControlParameter);
+        }
+
+            // Constructing jsonElementPeriodAlignment
+            JSONObject jsonElementPeriodAlignment = new JSONObject();
+            jsonElementPeriodAlignment.appendField(ConsentExtensionConstants.TITLE,
+                    ConsentExtensionConstants.CONTROL_PARAMETER_PERIOD_ALIGNMENT_TITLE);
+            String periodAlignment = (String) ((JSONObject) ((JSONArray) controlParameters
+                    .get(ConsentExtensionConstants.PERIODIC_LIMITS)).get(0))
+                    .get(ConsentExtensionConstants.PERIOD_ALIGNMENT);
+
+        if (periodAlignment != null) {
+            JSONArray periodAlignmentArray = new JSONArray();
+            periodAlignmentArray.add(periodAlignment);
+            jsonElementPeriodAlignment.appendField(ConsentExtensionConstants.DATA_SIMPLE, periodAlignmentArray);
+            consentDataJSON.add(jsonElementPeriodAlignment);
+        }
+
+            // Constructing jsonElementPeriodType
+            JSONObject jsonElementPeriodType = new JSONObject();
+            jsonElementPeriodType.appendField(ConsentExtensionConstants.TITLE,
+                    ConsentExtensionConstants.CONTROL_PARAMETER_PERIOD_TYPE_TITLE);
+            String periodType = (String) ((JSONObject) ((JSONArray) controlParameters
+                    .get(ConsentExtensionConstants.PERIODIC_LIMITS)).get(0)).get(ConsentExtensionConstants.PERIOD_TYPE);
+
+        if (periodType != null) {
+            JSONArray periodTypeArray = new JSONArray();
+            periodTypeArray.add(periodType);
+            jsonElementPeriodType.appendField(ConsentExtensionConstants.DATA_SIMPLE, periodTypeArray);
```

Comment:
Any reason we append same json value as in L640?

Reply:
Fixed in [fa29ae7]